src/Tests/UnitTest/TestComponentCSharp_Tests.cs

```diff
@@ -2253,96 +2253,145 @@ public void TestNonProjectedRuntimeClass()
+
+            // Test collection of composable with self-referential delegate
+            var composableWithDelegate = CreateObject<Composable>((obj) => {
+                obj.StringPropertyChanged += (Composable sender, string value) => Assert.Equal(sender, obj);
+            });
+            // TODO: implement IReferenceTracker to prevent delegate creating cycle
+            Assert.True(!IsCollectible(composableWithDelegate));
         }
-
-        // Manual for now - verify that all APIs targeting 19041 generate a warning
-        private void TestSupportedOSPlatformWarnings()
-        {
-            // Types
-            var a = new WarningAttribute();    // warning CA1416
-            Assert.NotNull(a);
-            var w = new WarningStruct{ i32 = 0 }; // warning CA1416
-            Assert.Equal(0, w.i32);     // warning CA1416
-            var v = WarningEnum.Value;
-            Assert.NotEqual(WarningEnum.WarningValue, v);   // warning CA1416
-
-            // Members
-            var o = new WarningClass();    // warning CA1416
-            o = new WarningClass(WarningEnum.Value);    // warning CA1416
-            o.WarningMethod();     // warning CA1416
-            var p = o.WarningProperty; // warning CA1416
-            o.WarningProperty = 0; // warning CA1416
-            p = o.WarningPropertySetter;
-            o.WarningPropertySetter = 0;   // warning CA1416
-            o.WarningEvent += (object s, Int32 v) => { }; // warning CA1416
-            o.WarningInterfaceMethod();     // warning CA1416
-            p = o.WarningInterfaceProperty; // warning CA1416
-            o.WarningInterfaceProperty = 0; // warning CA1416
-            p = o.WarningInterfacePropertySetter;
-            o.WarningInterfacePropertySetter = 0;   // warning CA1416
-            o.WarningInterfaceEvent += (object s, Int32 v) => { }; // warning CA1416
-
-            // Attributed statics
-            WarningStatic.WarningMethod(); // warning CA1416
-            WarningStatic.WarningProperty = 0; // warning CA1416
-            WarningStatic.WarningEvent += (object s, Int32 v) => { }; // warning CA1416
+#endif
+    }
+
+#if NET5_0
+    public class TestSupportedOSPlatformWarnings
+    {
+        [TestComponentCSharp.Warning]  // NO warning CA1416
+        class WarningManaged { };
+
+        class WarningSubclass : WarningClass
+        {
+            void InvokeOverridableWarnings()
+            {
+                WarningOverridableMethod(); // warning CA1416
+                WarningOverridableProperty = 0; // warning CA1416
+                // see https://github.com/microsoft/cppwinrt/issues/782
+                //WarningOverridableEvent += (object s, Int32 v) => { }; // warning CA1416
+            }
+        }
+
+        // Manual for now - verify that all APIs targeting 19041 generate a warning
+        // Consider invoking roslyn and scraping/analyzing output as an automated test
+        private void Test()
```

Comment:
Not a feedback for this PR, but as an FYI, we can probably automate this test this by creating a C# compilation context and triggering a compilation on it and then checking for diagnostics.  An example of this is the diagnostics test which Joshua wrote which do this.

Reply:
Feel free to open an issue with specific suggestion